src/menu.test.interactions.ts

```diff
@@ -233,6 +233,104 @@ it('remains open when a disabled link is clicked via `sendMouse()`', async () =>
   expect(target?.ariaExpanded).to.equal('true');
 });
 
+it('remains open when its options are clicked and the event is canceled', async () => {
+  const host = await fixture<GlideCoreMenu>(
+    html`<glide-core-menu open>
+      <button slot="target">Target</button>
+
+      <glide-core-menu-options>
+        <glide-core-menu-button label="Label"></glide-core-menu-button>
+        <glide-core-menu-link label="Label"></glide-core-menu-link>
+      </glide-core-menu-options>
+    </glide-core-menu>`,
+  );
+
+  // Wait for Floating UI.
+  await aTimeout(0);
+
+  host
+    .querySelector('glide-core-menu-options')
+    ?.addEventListener('click', (event: Event) => event.preventDefault());
+
+  await click(host.querySelector('glide-core-menu-button'));
+  await click(host.querySelector('glide-core-menu-link'));
+
+  const defaultSlot =
+    host?.shadowRoot?.querySelector<HTMLSlotElement>('slot:not([name])');
+
+  const target = host.querySelector('button');
+
+  expect(host.open).to.be.true;
+  expect(defaultSlot?.checkVisibility()).to.be.true;
+  expect(target?.ariaExpanded).to.equal('true');
+});
+
+it('remains open when its target is clicked and the event is canceled', async () => {
+  const host = await fixture<GlideCoreMenu>(
+    html`<glide-core-menu open>
+      <button slot="target">Target</button>
+
+      <glide-core-menu-options>
+        <glide-core-menu-button label="Label"></glide-core-menu-button>
+        <glide-core-menu-link label="Label"></glide-core-menu-link>
+      </glide-core-menu-options>
+    </glide-core-menu>`,
+  );
+
+  // Wait for Floating UI.
+  await aTimeout(0);
+
+  // Enter and Space are different than a mouse click in that they additionally produce
+  // a "click" on the option itself via the `#activeOption?.click()` in `#onSlotKeydown`.
+  // So we, like the consumer, have to cancel that event too to stop Menu from closing.
```

Comment:
I considered documenting this for consumers. But then I thought it's unlikely they'll need to keep Menu open when it's already open and its target is clicked. More likely they want to stop Menu from opening or else stop it from closing when a Menu Button or Menu Link is clicked.